### PR TITLE
OCPBUGS-17157: CatalogSource: add configurations for gRPC server memory footprint

### PR DESCRIPTION
**Pull request by stevekuznetsov**

### CODE REVIEW

File: pkg/operators/v1alpha1/catalogsource_types.go [hunk 1]
```diff
@@ -140,6 +141,21 @@ type GrpcPodConfig struct {
 	// +kubebuilder:validation:Enum=legacy;restricted
 	// +kubebuilder:default:=legacy
 	SecurityContextConfig SecurityConfig `json:"securityContextConfig,omitempty"`
+
+	// MemoryTarget configures the $GOMEMLIMIT value for the gRPC catalog Pod. This is a soft memory limit for the server,
+	// which the runtime will attempt to meet but makes no guarantees that it will do so. If this value is set, the Pod
+	// will have the following modifications made to the container running the server:
+	// - the $GOMEMLIMIT environment variable will be set to this value in bytes
+	// - the memory request will be set to this value
+	// - the memory limit will be set to 200% of this value
```
Comment (stevekuznetsov) [inline pkg/operators/v1alpha1/catalogsource_types.go:150]: Happy to remove this as well, unclear how folks feel about it.

Comment (perdasilva) [reply pkg/operators/v1alpha1/catalogsource_types.go:150]: I'm happy with that. Also learned some new stuff today: https://weaviate.io/blog/gomemlimit-a-game-changer-for-high-memory-applications

File: pkg/operators/v1alpha1/catalogsource_types.go [hunk 2]
```diff
@@ -140,6 +141,21 @@ type GrpcPodConfig struct {
...
+	//
+	// This field should be set if it's desired to reduce the footprint of a catalog server as much as possible, or if
+	// a catalog being served is very large and needs more than the default allocation. In order to determine a good
+	// approximation for this value, double the size of the package cache at /tmp/cache/cache/packages.json in the index
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply pkg/operators/v1alpha1/catalogsource_types.go:154]: Do we have proof that a cache-less image exists anywhere in the wild? Would be ideal to understand the lay of the land here and make simplifying assumptions as needed.

We can simply remove the default and make this entirely opt-in, if you wish.

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply pkg/operators/v1alpha1/catalogsource_types.go:154]: Removed the default.

**Comment by joelanford** [comment]
/approve

Looks like something to fix up in bindata though?

**Pull request merged**